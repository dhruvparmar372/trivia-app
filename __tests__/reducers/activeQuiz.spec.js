import {
  startQuiz as startQuizAction,
  endQuiz as endQuizAction,
  recordAnswer as recordAnswerAction
} from "source/actions/activeQuiz";
import activeQuizReducer, { initialState } from "source/reducers/activeQuiz";

const quizStartDate = "2017-12-31T18:30:00.000Z";
const dummyQuiz = {
  id: 1,
  startedOn: quizStartDate,
  questions: [
    {
      id: 1,
      type: "boolean",
      question: "1 + 1 = 2?",
      correct_answer: "True",
      incorrect_answers: ["False"]
    }
  ]
};

describe("activeQuiz reducer", () => {
  test("sets quiz data on startQuiz action", () => {
    expect(
      activeQuizReducer(initialState, startQuizAction({ id: 1 }, quizStartDate))
    ).toMatchSnapshot();
  });

  test("removes quiz data on endQuiz action", () => {
    expect(activeQuizReducer(dummyQuiz, endQuizAction())).toMatchSnapshot();
  });

  test("saves answer for quiz question on recordAnswer action", () => {
    expect(
      activeQuizReducer(
        dummyQuiz,
        recordAnswerAction(dummyQuiz.id, dummyQuiz.questions[0].id, {
          value: "True"
        })
      )
    ).toMatchSnapshot();
  });

  test("ignores answer if quizId does not match active quiz on recordAnswer action", () => {
    expect(
      activeQuizReducer(
        dummyQuiz,
        recordAnswerAction(dummyQuiz.id + 1, dummyQuiz.questions[0].id, {
          value: "True"
        })
      )
    ).toMatchSnapshot();
  });

  test("ignores answer if questionId does not match active quiz question on recordAnswer action", () => {
    expect(
      activeQuizReducer(
        dummyQuiz,
        recordAnswerAction(dummyQuiz.id, dummyQuiz.questions[0].id + 1, {
          value: "True"
        })
      )
    ).toMatchSnapshot();
  });
});
