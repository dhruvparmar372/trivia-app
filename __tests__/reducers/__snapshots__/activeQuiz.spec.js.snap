// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activeQuiz reducer ignores answer if questionId does not match active quiz question on recordAnswer action 1`] = `
Object {
  "id": 1,
  "questions": Array [
    Object {
      "correct_answer": "True",
      "id": 1,
      "incorrect_answers": Array [
        "False",
      ],
      "question": "1 + 1 = 2?",
      "type": "boolean",
    },
  ],
  "startedOn": "2017-12-31T18:30:00.000Z",
}
`;

exports[`activeQuiz reducer ignores answer if quizId does not match active quiz on recordAnswer action 1`] = `
Object {
  "id": 1,
  "questions": Array [
    Object {
      "correct_answer": "True",
      "id": 1,
      "incorrect_answers": Array [
        "False",
      ],
      "question": "1 + 1 = 2?",
      "type": "boolean",
    },
  ],
  "startedOn": "2017-12-31T18:30:00.000Z",
}
`;

exports[`activeQuiz reducer removes quiz data on endQuiz action 1`] = `null`;

exports[`activeQuiz reducer saves answer for quiz question on recordAnswer action 1`] = `
Object {
  "id": 1,
  "questions": Array [
    Object {
      "answer": Object {
        "value": "True",
      },
      "correct_answer": "True",
      "id": 1,
      "incorrect_answers": Array [
        "False",
      ],
      "question": "1 + 1 = 2?",
      "type": "boolean",
    },
  ],
  "startedOn": "2017-12-31T18:30:00.000Z",
}
`;

exports[`activeQuiz reducer sets quiz data on startQuiz action 1`] = `
Object {
  "id": 1,
  "startedOn": "2017-12-31T18:30:00.000Z",
}
`;
